---
title: "Transcriptional noise simulations - alignment accuracy"
output: html_notebook
---


```{r, fig.width=20,fig.height=11}

library(ggplot2)
library(dplyr)
library(cowplot)

#save(adf, file='txnoise_aln_stats.rda')
#
# load the aligner summary results
#noticed that hisat-nodta and hisat-rs give virtually identical results, so let us keep only hisat-nodta
#adf <- subset(adf, aligner != "hisat-rs")


load('txnoise_aln_stats.rda')


```
The reads were simulated from CHESS transcripts found in 3 GTEx tissues, 10 samples each (30 samples total).
Splicing noise was also added to the 30 samples (splice alignments were simulated from spurious multi-exon transcripts), which created another set of 30 samples.

Spliting the data into the "regular" data set (at) and the one with splicing noise added (ats):
```{r}
at <- subset(adf, grepl('_s\\d$', adf$sample)) 

## alignment data with splicing noise added:
ats <- subset(adf, grepl('\\.s$', adf$sample))

```


Percent of correct spliced alignments (multi-exon alignments)
Instead of data=at, we can exclude aligner entries by using:
 data=subset(at, aligner %in% c('hisat2','star','hisat2nodta','hisat21'))
 
```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=num_match_spl/num_sim_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of spliced reads correctly aligned") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.02)) +
        scale_x_continuous(breaks=seq(1, 30, 1))

```
When splicing noise is added, it looks pretty much the same for multi-exon reads, with STAR sensitivity (recall) still trailing in the 70s:

```{r, fig.width=11,fig.height=6}
ggplot(data=ats, aes(x=as.numeric(factor(sample)), y=num_match_spl/num_sim_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of spliced reads correctly aligned (splicing noise added)") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.02)) +
        scale_x_continuous(breaks=seq(1, 30, 1))
```

When considering all the reads - including the many single-exon, unspliced reads, STAR is catching up, raising its recall fraction into the 90s:

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=num_match/num_sim, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of reads correctly aligned") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.01)) +
        scale_x_continuous(breaks=seq(1, 30, 1))
```

All aligners reported some single-exon reads that were incorrectly mapped as spliced (false splice alignments).
Graph below shows the proportion of these false spliced mappings from the total number of spliced-aligned reads for each aligner
Essentially this is a false discovery rate for spliced alignments (junctions), FP/(FP+TP)

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=unspl_mapped_spl/num_aligned_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("splice alignments FDR") + 
        ggtitle("False Discovery Rate of spliced alignments") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) #+
        #scale_y_continuous(breaks=seq(50000, 900000, 50000))

```

When splicing noise is added the numbers are slightly higher, as expected:

```{r, fig.width=11,fig.height=6}
ggplot(data=ats, aes(x=as.numeric(factor(sample)), y=unspl_mapped_spl/num_aligned_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("splice alignments FDR") + 
        ggtitle("False Discovery Rate of spliced alignments (splicing noise added)") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) #+
        #scale_y_continuous(breaks=seq(50000, 900000, 50000))

```


On the other side, the aligners also report a number of multi-exon reads that were mistakenly aligned without introns (missed spliced alignments!):

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=spl_mapped_unspl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("multi-exon reads aligned without introns") + 
        ggtitle("Multi-exon reads incorrectly mapped as unspliced") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 18, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) +
        scale_y_continuous(breaks=seq(250000, 6000000, 500000))
        

```