---
title: "RNA-Seq simulations - alignment accuracy"
output: html_notebook
---

30 RNA-Seq samples were simulated with polyester. The reads were simulated from CHESS transcripts found in 3 GTEx tissues, 10 samples each.
These samples have intergenic and intronic "noise" transcripts added to the pool of "known" ("real") transcripts for the simulation.

Additional splicing noise was then added to these 30 simulated samples and the resulting sets, after shuffling, were aligned separately as an additional set of 30 samples.

* this R markdown notebook: [txnoise_aln_stats.Rmd](txnoise_aln_stats.Rmd)
* this notebook data: [txnoise_aln_stats.rda](txnoise_aln_stats.rda)

```{r, fig.width=20,fig.height=11}

library(ggplot2)
library(dplyr)

load('txnoise_aln_stats.rda') #loads the adf data frame object

```

Spliting the data into the "regular" data set (at) and the one with splicing noise added (ats):
```{r}
at <- subset(adf, grepl('_s\\d$', adf$sample)) 

## alignment data when splicing noise was added:
ats <- subset(adf, grepl('\\.s$', adf$sample))

```


Percentage of correct spliced alignments (multi-exon alignments). 

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=num_match_spl/num_sim_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of spliced reads correctly aligned") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.02)) +
        scale_x_continuous(breaks=seq(1, 30, 1))

```
When splicing noise is added, it looks pretty much the same for multi-exon reads, with STAR sensitivity (recall) still trailing in the 70s:

```{r, fig.width=11,fig.height=6}
ggplot(data=ats, aes(x=as.numeric(factor(sample)), y=num_match_spl/num_sim_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of spliced reads correctly aligned (splicing noise added)") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.02)) +
        scale_x_continuous(breaks=seq(1, 30, 1))
```

When considering all the reads - including the many single-exon, unspliced reads, STAR is catching up, raising its recall fraction into the 90s:

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=num_match/num_sim, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab(" proportion correctly aligned") + 
        ggtitle("Proportion of reads correctly aligned") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_y_continuous(breaks=seq(0.5, 0.98, 0.01)) +
        scale_x_continuous(breaks=seq(1, 30, 1))
```

All aligners reported some single-exon reads that were incorrectly mapped as spliced (false splice alignments).
Graph below shows the proportion of these false spliced mappings from the total number of spliced-aligned reads for each aligner
Essentially this is a false discovery rate for spliced alignments (junctions), FP/(FP+TP)

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=unspl_mapped_spl/num_aligned_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("splice alignments FDR") + 
        ggtitle("False Discovery Rate of spliced alignments") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) #+
        #scale_y_continuous(breaks=seq(50000, 900000, 50000))

```

When splicing noise is added the numbers are slightly higher, as expected:

```{r, fig.width=11,fig.height=6}
ggplot(data=ats, aes(x=as.numeric(factor(sample)), y=unspl_mapped_spl/num_aligned_spl, group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("splice alignments FDR") + 
        ggtitle("False Discovery Rate of spliced alignments (splicing noise added)") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) #+
        #scale_y_continuous(breaks=seq(50000, 900000, 50000))

```


On the other side, the aligners also report a number of multi-exon reads that were mistakenly aligned without introns
(missed spliced alignments, FN). Plotting the False Omission rate (FOR = FN/(FN+TN)). FOR in this case is the proportion of
these incorrectly unspliced alignments from the total number of unspliced alignments reported by each aligner

```{r, fig.width=11,fig.height=6}
ggplot(data=at, aes(x=as.numeric(factor(sample)), y=spl_mapped_unspl/(num_aligned-num_aligned_spl), group=aligner,
        color=aligner)) +  geom_line(size=1) + geom_point(aes(shape=aligner), size=2.4) + 
        xlab("samples") + ylab("multi-exon reads FOR") + 
        ggtitle("False omission rate of multi-exon reads incorrectly mapped as unspliced") +
        theme_minimal() +
        theme(plot.title = element_text(hjust = 0.5),
              axis.title.x = element_text(color="#404040", size=12, face="bold", margin=(margin(t=8))),
              axis.title.y = element_text(color="#404040", size=12, face="bold", margin=(margin(r=8)))
              ) +
        scale_shape_manual(values = c(0, 2, 20, 1)) +
        scale_x_continuous(breaks=seq(1, 30, 1)) 
```